SoundWire/Intel: update maximum number of links

plbossart opened the pull request:
we can have up to 5 links in newer platforms.

plbossart committed:
ALSA/ASoC/SoundWire: Intel: use single definition for SDW_INTEL_MAX_LINKS

The definitions are currently duplicated in intel-sdw-acpi.c and
sof_sdw.c.  Move the definition to the sdw_intel.h header, and change
the prefix to make it Intel-specific.

No functionality change in this patch.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
soundwire: intel: increase maximum number of links

Intel platforms have enabled 4 links since the beginning, newer
platforms now have 5 links. Update the definition accordingly.

This patch will have no effect on older platforms where the number of
links was hard-coded. A follow-up patch will add a dynamic check that
the ACPI-reported information is aligned with hardware capabilities on
newer platforms.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

ujfalusi commented on include/linux/soundwire/sdw_intel.h:
```diff
@@ -450,6 +450,6 @@ extern const struct sdw_intel_hw_ops sdw_intel_lnl_hw_ops;
  * Max number of links supported in hardware
  */
-#define SDW_INTEL_MAX_LINKS                4
+#define SDW_INTEL_MAX_LINKS                5
```
I guess it is OK to accept 5 links on older platforms which only supports 4?
Iow, have ever seen cases that `Link count 5 exceeds max 4` is printed?

plbossart replied:
older platforms are hard-coded to 4, so there's no risk.

But for newer platforms, this could vary over time, so we need to check that the ACPI reported information is aligned with the LCAP register that can be read at probe time.

will add a new patch.

plbossart committed:
soundwire: intel: add probe-time check on link id

In older platforms, the number of links was constant and hard-coded to
4. Newer platforms can have varying number of links, so we need to add
a probe-time check to make sure the ACPI-reported information with
_DSD properties is aligned with hardware capabilities reported in the
SoundWire LCAP register.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart commented:
LNL RVP HDA alsa-bat playback issue https://sof-ci.01.org/linuxpr/PR5108/build4067/devicetest/index.html?model=LNLM_RVP_HDA&testcase=check-alsabat-headset-playback-821

This problem cannot possibly be related to this SoundWire PR, so merging.
